lib/Request.js

```diff
@@ -112,6 +117,76 @@ Request.prototype.getQuery = function(query) {
     return this._url.query.hasOwnProperty(query) ? this._url.query[query] : undefined;
 };
 
+/**
+ * Validates the query with the provided schema.
+ * Query parameters that are not included in the schema are ignored.
+ * Query parameters that are defined as type "integer" or "number" are converted accordingly.
+ * Returns a validation object with two properties:
+ *   - boolean valid -- true if the query parameters are valid
+ *   - array errors -- array of properties that are in error, if any
+ *
+ * @param object schema - a schema in json-schema format.
+ * @return object - validation result
+ */
+Request.prototype.validateQuery = function(schema) {
+    // Nothing to validate if the schema is not an object (rare)
+    if (!schema.properties) return true;
+
+    var query = this.getQueries();
+
+    // Validate query with schema
+    var validate = jsen(schema);
+
+    // _.forOwn(query, function(value, key) {
+    for (var key in query) {
+        // Skip prototype properties, if any
+        if (!query.hasOwnProperty(key)) continue;
+
+        // Skip properties that are not in the schema
+        if (!schema.properties.hasOwnProperty(key)) continue;
```

Comment:
So we ignore properties that aren't in the schema? Nice, this'll take care of the "props" query string parameter to filter what to get from the resource.